src/Space.test.ts

```diff
@@ -61,12 +62,14 @@ describe('Space', () => {
 
   describe('enter', () => {
     it<SpaceTestContext>('enter a space successfully', async ({ space, presence }) => {
+      mockSubscribeForSpaceEnter(presence);
```

Comment:
I'd propose we do this a bit differently. I think having ` mockSubscribeForSpaceEnter` in different tests will just make it harder to decide if that method should be used or the alternative `presenceMap` [approach](https://github.com/ably/spaces/blob/main/src/Locks.test.ts#L30-L32), which sets the presence set, without needing to call `enter()`.

So my suggestion would be to for this spec (that tests `enter`) use a `beforeEach` that sets up the mock for presence, but all the other specs are refactored to not call `enter()` but just use the `presenceMap` approach. WDYT?

Reply:
It sounds reasonable. I'll have to take a bit of time to understand how the `presenceMap` is currently used, which I'll do now.

Reply:
Turns out that none of these `space.enter()` calls in the other tests were needed, so I've removed them.

Reply:
(see ce44eb964d4694ff71675643c8310e26b645a485)

Reply:
I've moved the mocking of `presence.subscribe` into the `beforeEach` of `enter`’s tests, as you suggested.